### PR TITLE
chore(indents): use node:id() instead of node_fmt

### PR DESCRIPTION
**Pull request by theHamsta**
@kyazdani42 I think the code should be roughly equivalent except that `:id()` doesn't have the nil check. It appears that the outer code already checks for `nil`

### CODE REVIEW

File: lua/nvim-treesitter/indent.lua [hunk 1]
```diff
@@ -107,13 +102,13 @@ function M.get_indent(lnum)
 
   while node do
```
Comment (theHamsta) [inline lua/nvim-treesitter/indent.lua:103]: node ~=nil

File: lua/nvim-treesitter/indent.lua [hunk 2]
```diff
@@ -93,12 +88,12 @@ function M.get_indent(lnum)
   if not node then
     local wrapper = root:descendant_for_range(lnum-1, 0, lnum-1, -1)
     node = wrapper:child(0) or wrapper
```
Comment (theHamsta) [inline lua/nvim-treesitter/indent.lua:90]: `wrapper ~= nil`

File: lua/nvim-treesitter/indent.lua [hunk 3]
```diff
@@ -93,12 +88,12 @@ function M.get_indent(lnum)
   if not node then
     local wrapper = root:descendant_for_range(lnum-1, 0, lnum-1, -1)
```
Comment (theHamsta) [inline lua/nvim-treesitter/indent.lua:89]: root ~= nil

File: lua/nvim-treesitter/indent.lua [hunk 4]
```diff
@@ -73,7 +68,7 @@ function M.get_indent(lnum)
       end
 
       -- nodes can be marked @return to prevent using them
-      if prev_node and not q.returns[node_fmt(prev_node)] then
+      if prev_node and not q.returns[prev_node:id()] then
```
Comment (theHamsta) [inline lua/nvim-treesitter/indent.lua:71]: prev_node ~= nil

File: lua/nvim-treesitter/indent.lua [hunk 5]
```diff
@@ -16,17 +16,12 @@ local function get_node_at_line(root, lnum)
...
-  return tostring(node)
-end
-
 local get_indents = tsutils.memoize_by_buf_tick(function(bufnr, root, lang)
   local get_map = function(capture)
     local matches = queries.get_capture_matches(bufnr, capture, 'indents', root, lang) or {}
     local map = {}
     for _, node in ipairs(matches) do
-      map[tostring(node)] = true
+      map[node:id()] = true
```
Comment (theHamsta) [inline lua/nvim-treesitter/indent.lua:24]: node ~= nil

**Comment by kyazdani42** [comment]
is this ready for review ? i see you've left some comments with `~= nil`

**Comment by theHamsta** [comment]
@kyazdani42 I just wanted to document that it should still be correct and equivalent to the former code

**Comment by kyazdani42** [comment]
i will run the tests and if it works like before I'll merge this then

**Comment by kyazdani42** [comment]
running the tests shows diff of regression :/

**Comment by theHamsta** [comment]
Then better leave it the old way

**Pull request closed without merging**